### PR TITLE
Add Inliner CID Builder.

### PR DESCRIPTION
**Pull request by kevina**
Will be used by https://github.com/ipfs/go-ipfs/pull/5281.

CC @Stebalien

### CODE REVIEW

File: inliner.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import (
+	cid "github.com/ipfs/go-cid"
+	mhash "github.com/multiformats/go-multihash"
+)
+
+// Inliner is a cid.Builder that will use the id multihash when the
+// size of the content is no more than limit
+type Inliner struct {
+	cid.Builder
+	Limit int
```
Comment (Stebalien) [inline inliner.go]: Should probably be documented (inclusive? exclusive?).

**Review by Stebalien — APPROVED** [review]
I'd call this `InlineBuilder` or `InliningBuilder` to be a bit more descriptive but that's up to you.

This really is elegant.

File: inliner.go (hunk 1 above)
Comment (kevina) [reply inliner.go]: Do you mean "Limit"? I said above "size of the content is no more than limit" which should imply inclusive.

Comment (Stebalien) [reply inliner.go]: Ah. I didn't see that. Yes, that should be sufficient.

**Comment by Stebalien** [comment]
I'm still not in love with the name but feel free to merge when ready. LGTM.

**Comment by kevina** [comment]
@Stebalien Changed the name to InlineBuilder.

Merging.

**Pull request merged**